internal/mode/static/nginx/config/servers.go

```diff
@@ -30,15 +30,25 @@ var baseHeaders = []http.Header{
 		Name:  "X-Forwarded-For",
 		Value: "$proxy_add_x_forwarded_for",
 	},
-	{
+}
+
+// httpBaseHeaders contains the constant headers set in each HTTP server block
+var httpBaseHeaders = append(baseHeaders,
```

Comment:
constructing  httpBaseHeaders and grpcBaseHeaders works. But at the same time, some similar code will not work. For example -- https://go.dev/play/p/CEoyHhUjhit

If append doesn't allocate a new backing array, the code will not work.

I suggest avoiding appending to the same slice all together, so that we don't try to do it in any other places where it might not work as we expect.